include/simdjson/stringparsing_common.h

```diff
@@ -0,0 +1,91 @@
+// This file contains the common code every implementation uses
+// It is intended to be included multiple times and compiled multiple times
+// We assume the file in which it is include already includes
+// "simdjson/stringparsing.h" (this simplifies amalgation)
+
+#ifdef TARGETED_ARCHITECTURE
```

Comment:
Given that it's an error to include these files without setting TARGETED_ARCHITECTURE, I'd treat this more like argument validation and switch it to #ifndef / #error. This is in the nitpick arena (isn't a bug, and is just a defensive measure), so feel free to ignore :)

```suggestion
#ifndef TARGETED_ARCHITECTURE
#error TARGETED_ARCHITECTURE must be set before including this file.
#endif
```
Then the #endif at the end of the file wouldn't be needed. Not sure how to add that part as a suggestion here.